programs/stake/src/stake_state.rs

```diff
@@ -6666,6 +6666,120 @@ mod tests {
         }
     }
 
+    /// Ensure that `split()` correctly handles prefunded destination accounts.  When a destination
+    /// account already has funds, ensure the minimum split amount reduces accordingly.
+    #[test]
+    fn test_split_destination_minimum_stake_delegation() {
+        let rent = Rent::default();
+        let rent_exempt_reserve = rent.minimum_balance(std::mem::size_of::<StakeState>());
+        let source_pubkey = Pubkey::new_unique();
+        let source_meta = Meta {
+            rent_exempt_reserve,
+            ..Meta::auto(&source_pubkey)
+        };
+        // Set the source's starting balance to something large to ensure its post-split
+        // balance meets all the requirements
+        let source_starting_balance = u64::MAX;
+        let source_account = AccountSharedData::new_ref_data_with_space(
+            source_starting_balance,
+            &StakeState::Initialized(source_meta),
+            std::mem::size_of::<StakeState>(),
+            &id(),
+        )
+        .unwrap();
+        let source_keyed_account = KeyedAccount::new(&source_pubkey, true, &source_account);
+
+        for (dest_starting_balance, split_amount, expected_result) in [
+            // split amount must be non zero
+            (
+                rent_exempt_reserve + MINIMUM_STAKE_DELEGATION,
+                0,
+                Err(InstructionError::InsufficientFunds),
+            ),
+            // any split amount is OK when destination account is already fully funded
+            (rent_exempt_reserve + MINIMUM_STAKE_DELEGATION, 1, Ok(())),
+            // if destination is only short by 1 lamport, then split amount can be 1 lamport
+            (
+                rent_exempt_reserve + MINIMUM_STAKE_DELEGATION - 1,
+                1,
+                Ok(()),
+            ),
+            // dest short by 1 lamport, so 0 isn't enough
+            (
+                rent_exempt_reserve + MINIMUM_STAKE_DELEGATION - 1,
+                0,
+                Err(InstructionError::InsufficientFunds),
+            ),
```

Comment:
Is this test-case obviated by `// split amount must be non zero`?

Reply:
Yes. I had it in for completeness with the pattern of cases I established, but I've removed it since there's no gain (and potentially confusing).